### PR TITLE
feat: pipeline job support network mode

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
pipeline job support network mode

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=362555&iterationID=1580&type=TASK)

#### Specified Reviewers:

/assign @sfwn @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  Support pipeline job support network mode             |
| 🇨🇳 中文    |     实现流水线普通任务支持配置容器网络         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by iutx — APPROVED** [review]
/lgtm

File: internal/tools/pipeline/pipengine/actionexecutor/plugins/k8sjob/k8sjob.go [hunk 1]
```diff
@@ -486,6 +487,7 @@ func (k *K8sJob) generateKubeJob(specObj interface{}, clusterInfo apistructs.Clu
 					Volumes:               vols,
 					EnableServiceLinks:    func(enable bool) *bool { return &enable }(false),
 					ShareProcessNamespace: func(b bool) *bool { return &b }(false),
+					HostNetwork:           isHostNetwork,
```
Comment (iutx) [inline internal/tools/pipeline/pipengine/actionexecutor/plugins/k8sjob/k8sjob.go:490]: kubeJob path: Spec.Template.Spec.DNSPolicy

Need to be set corev1.DNSClusterFirstWithHostNet when `HostNetwork` is `true`. @chengjoey

Comment (chengjoey) [reply internal/tools/pipeline/pipengine/actionexecutor/plugins/k8sjob/k8sjob.go:490]: > kubeJob path: Spec.Template.Spec.DNSPolicy
> 
> Need to be set corev1.DNSClusterFirstWithHostNet when `HostNetwork` is `true`. @chengjoey

has DNSPolicy set to `corev1.DNSClusterFirstWithHostNet` been successful for the cdp task?
If the verification is successful, I will submit the changes in this pr

File: apistructs/job.go [hunk 2]
```diff
@@ -111,6 +111,7 @@ type JobFromUser struct {
 	BackoffLimit            int                    `json:"backoffLimit,omitempty"`
 	Params                  map[string]interface{} `json:"params,omitempty"`
 	TaskContainers          []TaskContainer        `json:"taskContainers"`
+	Network                 map[string]string      `json:"network,omitempty"`
```
Comment (sfwn) [inline apistructs/job.go:114]: Use typed struct is better. Or you need to clarify all the selectable values.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/job.go (hunk 2 above)
Comment (chengjoey) [reply apistructs/job.go]: refactor network to type, and impl `IsHostMode` for k8s executor

**Review by sfwn — APPROVED** [review]

**Pull request merged**